jcommune-service/src/test/java/org/jtalks/jcommune/service/transactional/TransactionalAuthenticatorTest.java

```diff
@@ -198,11 +201,32 @@ public void authenticateUserWithNewCredentialsShouldBeSuccessful() throws Except
                 .thenThrow(new BadCredentialsException(null)).thenReturn(expectedToken);
         preparePlugin(oldUser.getUsername(), passwordHash, authInfo);
 
-        boolean result = authenticator.authenticate(loginUserDto, httpRequest, httpResponse);
+        AuthenticationStatus result = authenticator.authenticate(loginUserDto, httpRequest, httpResponse);
 
         verify(userDao).saveOrUpdate(oldUser);
 
-        assertTrue(result, "Authentication user with new credentials should be successful.");
+        assertEquals(result, AuthenticationStatus.AUTHENTICATED,
+                "Authentication user with new credentials should be successful.");
+    }
+
+    @Test
+    public void authenticateNotEnabledUserShouldFail() throws Exception {
+            String password = "password";
+            String passwordHash = "5f4dcc3b5aa765d61d8327deb882cf99";
+            LoginUserDto loginUserDto = createDefaultLoginUserDto();
+            JCUser oldUser = prepareOldUser(loginUserDto.getUserName());
+            when(userDao.getByUsername(oldUser.getUsername())).thenReturn(oldUser);
```

Comment:
Do you think maybe there is a way shortening the test method and removing extra when()?